tpp/src/Rpc/UnauthenticatedClient.cs

```diff
@@ -66,5 +75,82 @@ public Task<Proto.Common.TokenProtos.TokenRequest> RetrieveTokenRequest(string t
             return gateway.RetrieveTokenRequestAsync(request)
                 .ToTask(response => response.TokenRequest);
         }
+
+        /// <summary>
+        /// Recovers an eIDAS-verified member with eidas payload.
+        /// 
+        /// </summary>
+        /// <param name="payload">a payload containing member id, the certificate and a new key to add to the member</param>
+        /// <param name="signature">a payload signature with the private key corresponding to the certificate</param>
+        /// <param name="cryptoEngine">a crypto engine that must contain the privileged key that is included in
+        ///      the payload(if it does not contain keys for other levels they will be generated)</param>
+        /// <returns>a task of a new member</returns>
+        public Task<ProtoMember> RecoverEidasMember(
+            EidasRecoveryPayload payload,
+            string signature,
+            ICryptoEngine cryptoEngine)
+        {
+            Key privilegedKey = payload.Key;
+            Key standardKey = GetOrGenerateKeyForLevel(cryptoEngine,
+                Level.Standard);
+            Key lowKey = GetOrGenerateKeyForLevel(cryptoEngine,
+                Level.Low);
+
+            IList<Key> keys = new List<Key> {privilegedKey, standardKey, lowKey};
+            // TODO(RD-3764): createSigner by keyId (to make sure it's for the key in the payload)
```

Comment:
RD-3764 is completed, so you can create signer by keyId

Reply:
done